### PR TITLE
add workflow name to conclusions dictionary in trymerge

### PR DESCRIPTION
**Pull request by clee2000**
Xla hash updates were failing because the merge rules json was looking for the name of the job in the format `<workflow name> / <job name>` but try merge just has `<job name>`.  

An easier solution would be to just get rid of pull in the merge rules but I think it would be better to keep to this naming convention.

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
Why did thus affect only XLA jobs?

File: .github/scripts/trymerge.py [hunk 1]
```diff
@@ -967,12 +970,15 @@ def add_conclusions(edges: List[Dict[str, Dict[str, Any]]]) -> None:
             checkruns = node["checkRuns"]
             if workflow_run is not None:
                 conclusions[workflow_run["workflow"]["name"]] = (node["conclusion"], node["url"])
+            check_run_name_prefix = "" if workflow_run is None else f'{workflow_run["workflow"]["name"]} / '
```
Comment (zengk95) [inline .github/scripts/trymerge.py:973]: Can we extract this out to a method? It's repeating some stuff and it's a bit hard to parse.

**Comment by clee2000** [comment]
it only affects xla because xla is the only one with a merge rules that specifies which builds + tests it wants, which is needed to prevent us from accidentally merging hash updates when xla is disabled

**Review by zengk95 — APPROVED** [review]
lgtm. I don't hate the other solution you had though.

File: .github/scripts/trymerge.py (hunk 1 above)
Comment (huydhn) [reply .github/scripts/trymerge.py:973]: +1 I confuse myself looking at the 2 "almost" identical `add_conclusions` functions in `get_checkrun_conclusions` and `get_land_checkrun_conclusions`. It's good to have them refactored into one. It can be in another PR though because it's out of scope for this one

**Review by huydhn — APPROVED** [review]
